externals/llvm-external-projects/torch-mlir-dialects/lib/Dialect/Tcp/IR/TcpOps.cpp

```diff
@@ -155,4 +156,43 @@ LogicalResult ConcatOp::verify() {
   return success();
 }
 
+LogicalResult CastOp::verify() {
+  auto inputType = getIn().getType().cast<RankedTensorType>();
+  auto outputType = getOut().getType().cast<RankedTensorType>();
+
+  if (inputType.getElementType().isa<FloatType>()) {
+    if (getInIntSignedness())
+      return emitOpError(
+          "in_int_signedness attr should not set when input is FP");
+  }
+
+  if (inputType.getElementType().isa<IntegerType>()) {
+    if (!getInIntSignedness())
+      return emitOpError(
+          "in_int_signedness attr must be set when input is INT");
+    if (inputType.getElementType().isInteger(1) &&
+        getInIntSignedness().value() != Signedness::Signless)
+      return emitOpError("in_int_signedness attr must be set to "
+                         "Signedness::Signless when input is i1");
+  }
+
+  if (outputType.getElementType().isa<FloatType>()) {
+    if (getOutIntSignedness())
+      return emitOpError(
+          "out_int_signedness attr should not set when output is FP");
+  }
+
+  if (outputType.getElementType().isa<IntegerType>()) {
+    if (!getOutIntSignedness())
+      return emitOpError(
+          "out_int_signedness attr must be set when output is INT");
+    if (outputType.getElementType().isInteger(1) &&
+        getOutIntSignedness().value() != Signedness::Signless)
+      return emitOpError("out_int_signedness attr must be set to "
+                         "Signedness::Signless when output is i1");
+  }
+
+  return success();
```

Comment:
What if inputType and/or outputType aren't integer or floating-point tensors? As it's currently written, the verifier will succeed but that's probably undesirable?